guides/basics/introduction-to-avalonia-concepts.md

```diff
@@ -0,0 +1,3686 @@
+
+You can see that the solution consists of 3 projects: the main project NP.Demos.XamlNamespacesSample and two projects on which the main project depends: Dependency1Proj and Dependency2Proj:
+
+![](../../.gitbook/assets/sample1.png)
+
+Compile and run the solution - here is what you are going to see:
+
+![](../../.gitbook/assets/sample2.png)
+
+There are 5 square buttons of different colors stacked vertically within a window. 
+
+Here is the relevant code of MainWindow.xaml file:
+
+```markup
+<Window xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:dep1="clr-namespace:Dependency1Proj;assembly=Dependency1Proj"
+        xmlns:dep1_sub_Folder="clr-namespace:Dependency1Proj.SubFolder;assembly=Dependency1Proj"
+        xmlns:local="clr-namespace:NP.Demos.XamlNamespacesSample"
+        xmlns:dep2="https://avaloniademos.com/xaml"
+        ...>
+  <StackPanel HorizontalAlignment="Center"
+              VerticalAlignment="Center">
+    <dep1:RedButton/>
+    <dep1_sub_Folder:BlueButton/>
+    <local:BrownButton/>
+    <dep2:GreenButton/>
+    <dep2:GrayButton/>
+  </StackPanel>
+</Window>
+
+```
+
+There are four custom namespaces defined by the following lines of the top XML Tag:
+
+```markup
+xmlns:dep1="clr-namespace:Dependency1Proj;assembly=Dependency1Proj"
+xmlns:dep1_test_Folder="clr-namespace:Dependency1Proj.SubFolder;assembly=Dependency1Proj"
+xmlns:local="clr-namespace:NP.Demos.XamlNamespacesSample"
+xmlns:dep2="https://avaloniademos.com/xaml"
+```
+
+Different buttons are are referred two by the corresponding XAML namespace prefixes. Let us take a look at `<RedButton/>` `RedButton` class is defined as a C\# class under Dependency1Proj project. Here is its code:
+
+```csharp
+...
+namespace Dependency1Proj
+{
+    public class RedButton : Button, IStyleable
+    {
+        Type IStyleable.StyleKey => typeof(Button);
+
+        public RedButton()
+        {
+            Background = new SolidColorBrush(Colors.Red);
+            Width = 30;
+            Height = 30;
+        }
+    }
+}
+```
+
+The line
+
+```csharp
+Type IStyleable.StyleKey => typeof(Button);
+```
+
+\(and the fact that `RedButton` class implements `IStyleable` interface\) ensures that the default button Styles from the main theme will also be applied to the class `RedButton` which derives from Avalonia `Button` class. The constructor of the button assigns the button color to red and sets the button to have height and width of 30 generic pixels.
+
+Note that the code for each of the buttons of the sample is exactly the same as that for `RedButton` aside from the button class name, C\# namespace and the color assigned to the button.
+
+Now take a look at the line that defines the XAML namespace prefix `dep1` by which we refer to this button inside the XAML file:
+
+```markup
+xmlns:dep1="clr-namespace:Dependency1Proj;assembly=Dependency1Proj"
```

Comment:
afaik we have simpler syntax to define this,it would be more readable for newbies